### PR TITLE
Schema cleanup and bugfixes

### PR DESCRIPTION
**Pull request by david-waltermire-nist**
# Committer Notes

This set of changes are to cleanup a variety schemas.

- Fixes a bug in the base64 data type that caused by most of the allowed alphabet to be excluded as well as padding characters.
- Corrected duration-based patterns to enforce the correct semantics.
- Simplified boolean patterns.
- Aligned integer and string data type derivations.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.

### CODE REVIEW

**Review by wendellpiez — APPROVED** [review]
Thanks @david-waltermire-nist

**Review by aj-stein-nist — DISMISSED** [review]

File: schema/json/metaschema-datatypes.json
Comment (nikitawootten-nist) [inline schema/json/metaschema-datatypes.json:9]: B: Which flavors of Regex are we targeting (a readme or ADR documenting this would be helpful).

This regex pattern is invalid in PRCRE derived flavors.

**Review by nikitawootten-nist — CHANGES_REQUESTED** [review]
A few questions below :)

File: schema/json/metaschema-datatypes.json
Comment (david-waltermire-nist) [reply schema/json/metaschema-datatypes.json:9]: This should fix the regex, which should still work well with ECMA.

```suggestion
      "pattern": "^[0-9A-Za-z+\/]+={0,2}$",
```

ECMA regex is required here, since it is JSON. In general we are targeting PCRE, Java, ECMA, XMLSchema flavors. I agree that an ADR on this would be useful. Creating an ADR is beyond the scope of this PR. Maybe an issue can be created for this?

**Comment by aj-stein-nist** [comment]
Is it ok I request a rebase against current develop to include #344 to make testing easier? I work around it, but it would be minorly helpful. :-)

**Comment by david-waltermire-nist** [comment]
> Is it ok I request a rebase against current develop to include #344 to make testing easier? I work around it, but it would be minorly helpful. :-)

Done.

**Comment by david-waltermire-nist** [comment]
I believe this PR is ready to merge. I'll merge it once @nikitawootten-nist approves.

**Review by nikitawootten-nist — APPROVED** [review]

**Pull request merged**
